enroll/transport_http.go

```diff
@@ -0,0 +1,109 @@
+package enroll
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/fullsailor/pkcs7"
+	httptransport "github.com/go-kit/kit/transport/http"
+	"github.com/groob/plist"
+)
+
+type HTTPHandlers struct {
+	EnrollHandler          http.Handler
+	OTAEnrollHandler       http.Handler
+	OTAPhase2Phase3Handler http.Handler
+}
+
+func MakeHTTPHandlers(ctx context.Context, endpoints Endpoints, opts ...httptransport.ServerOption) HTTPHandlers {
+	h := HTTPHandlers{
+		EnrollHandler: httptransport.NewServer(
+			endpoints.GetEnrollEndpoint,
+			decodeMDMEnrollRequest,
+			encodeResponse,
+			opts...,
+		),
+		OTAEnrollHandler: httptransport.NewServer(
+			endpoints.OTAEnrollEndpoint,
+			nilRequest,
+			encodeResponse,
+			opts...,
+		),
+		OTAPhase2Phase3Handler: httptransport.NewServer(
+			endpoints.OTAPhase2Phase3Endpoint,
+			decodeOTAPhase2Phase3Request,
+			encodeResponse,
+			opts...,
+		),
+	}
+	return h
+}
+
+func nilRequest(_ context.Context, _ *http.Request) (interface{}, error) {
+	return nil, nil
+}
+
+func decodeMDMEnrollRequest(_ context.Context, r *http.Request) (interface{}, error) {
+	switch r.Method {
+	case "GET":
+		return mdmEnrollRequest{}, nil
+	case "POST":
+		data, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			return nil, err
+		}
+		p7, err := pkcs7.Parse(data)
+		if err != nil {
+			return nil, err
+		}
+		// TODO: We should verify but not currently possible. Apple
+		// does no provide a cert for the CA.
+		var request depEnrollmentRequest
+		if err := plist.Unmarshal(p7.Content, &request); err != nil {
+			return nil, err
+		}
+		return request, nil
+	default:
+		return nil, errors.New("unknown enrollment method")
+	}
+}
+
+func encodeResponse(ctx context.Context, w http.ResponseWriter, response interface{}) error {
+	switch resp := response.(type) {
```

Comment:
Let's drop the type switch here. IMO it's not necessary. 
If you'd like to keep it, I'd actually prefer a panic over an error here since it's a programmer error. 

Finally, instead of doing 
```
switch resp := ....
_ = resp
``` 
to avoid the unused variable error, you can do `switch resp.(type) {` without declaring a var.